testutil/utils.go

```diff
@@ -14,10 +14,14 @@ import (
 
 func RandomValidNamespace() namespace.ID {
 	for {
-		s := tmrand.Bytes(8)
-		if bytes.Compare(s, appconsts.MaxReservedNamespace) > 0 {
-			return s
+		ns := tmrand.Bytes(8)
+		isReservedNS := bytes.Compare(ns, appconsts.MaxReservedNamespace) <= 0
+		isParityNS := bytes.Equal(ns, appconsts.ParitySharesNamespaceID)
+		isTailPaddingNS := bytes.Equal(ns, appconsts.TailPaddingNamespaceID)
+		if isReservedNS || isParityNS || isTailPaddingNS {
+			continue
 		}
+		return ns
```

Comment:
[question] Nice! There are multiple implementations of this function that may benefit from similar parity / tail padding namespace checks:

1. https://github.com/rootulp/celestia-app/blob/2f26c385831048808d99f4626aa3569df18c187c/app/test_util.go#L178-L185
2. https://github.com/rootulp/celestia-app/blob/2f26c385831048808d99f4626aa3569df18c187c/testutil/payment/testutil.go#L134-L141
3. https://github.com/rootulp/celestia-app/blob/2f26c385831048808d99f4626aa3569df18c187c/app/test/block_size_test.go#L299-L306
4. https://github.com/rootulp/celestia-app/blob/2f26c385831048808d99f4626aa3569df18c187c/pkg/prove/proof_test.go#L249-L257

It's not clear to me if these should be refactored into one function definition. Were they intentionally duplicated to avoid cyclic imports / importing across package boundaries?